src/varity/ref_gene.clj

```diff
@@ -137,6 +138,73 @@
        (some #(in-exon? pos %))
        (true?)))
 
+(defn tx-region
+  "Returns a genomic region of the given gene."
+  [{:keys [chr tx-start tx-end strand]}]
+  {:chr chr, :start tx-start, :end tx-end, :reverse? (= strand "-")})
+
+(defn cds-region
+  "Returns a genomic region of a coding sequence of the given gene. Returns nil
+  if the gene is a non-coding RNA."
+  [{:keys [chr cds-start cds-end strand]}]
+  (when (<= cds-start cds-end)
+    {:chr chr, :start cds-start, :end cds-end, :reverse? (= strand "-")}))
+
+(defn exon-seq
+  "Returns a lazy sequence of regions corresponding to each exon in a gene. The
+  exons are ordered by their index, thus they're reversed in genomic coordinate
+  if the gene is on the reverse strand."
+  [{:keys [chr strand exon-ranges]}]
+  (let [exon-count (count exon-ranges)
+        reverse? (= strand "-")]
+    (->> exon-ranges
+         ((if reverse? reverse identity))
+         (map-indexed
+          (fn [i [s e]]
+            {:exon-index (inc i), ;; 1-origin
+             :exon-count exon-count,
+             :chr chr,
+             :start s,
+             :end e,
+             :reverse? reverse?})))))
+
+(defn cds-exon-seq
```

Comment:
I think `cds-seq` is correct name. `cds-exon-` is wordy because CDS does not include introns.